doc/md/fundamentals/2-types/3-functions.md

```diff
@@ -4,25 +4,25 @@ sidebar_position: 3
 
 # Function types
 
-Functions can be used to define abstract code as reusable building blocks.  
+Functions are reusable chunks of code that perform a specific task. A function is defined with a name, optional parameters, and returns a result of a defined when called. A function can also specify a return type for the value it produces.
 
-Functions are parameterized bits of code that produce results when supplied with arguments. In the process of producing a result, the execution of the function can have other side-effects, like modifying state, printing to the log, or sending messages to other canisters. 
+:::info Side Effects
+In the process of producing a result, the execution of the function can have other side-effects, like modifying state, printing to the log, or sending messages to other canisters.
+:::
 
-Functions can be synchronous or asynchronous. A synchronous function blocks the caller until the function returns with its result, allowing the caller to proceed. This is the usual notion of function implemented by other programming languages.  
+Functions can be synchronous or asynchronous. A synchronous function blocks the caller until it returns with a result, just like functions in most traditional programming languages.
 
-An asynchronous functions immediately returns to its caller. Its execution is decoupled from the execution of the caller. Instead, the caller may receive a **future** value, which acts as a placeholder for the result of the call. This future is replaced with a result when the function call returns. 
+An asynchronous function returns immediately, providing a **future** value as a placeholder for its result. The caller can await the future later to retrieve the result or ignore it and continue with other tasks.
 
-The caller can retrieve the result by awaiting the future at a later point in time. It can also ignore the result entirely by never awaiting the future. Between calling the asynchronous function and awaiting its result, the caller is able to proceed with other tasks.  
+Motoko has functions with different capabilities.
+Local functions, declared using just the `func` keyword, are typically synchronous but can be asynchronous if their body is an `async` expression.
```

Comment:
Local functions can only be called inside the actor - they are not available to remote callers and cannot be shared in messages.